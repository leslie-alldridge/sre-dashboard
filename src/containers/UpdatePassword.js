import React, { Component } from "react";
import { Redirect } from "react-router-dom";
import TextField from "@material-ui/core/TextField";
import axios from "axios";

import {
  LinkButtons,
  SubmitButtons,
  HeaderBar,
  homeButton,
  cancelButton,
  saveButton,
  loginButton,
  inputStyle
} from "../components";

const loading = {
  margin: "1em",
  fontSize: "24px"
};

const title = {
  pageTitle: "Update Password Screen"
};

class UpdatePassword extends Component {
  constructor(props) {
    super(props);

    this.state = {
      username: "",
      password: "",
      loadingUser: false,
      updated: false,
      error: false
    };
  }

  componentDidMount() {
    this.setState({ loadingUser: true });

    let accessString = localStorage.getItem("JWT");
    if (accessString === null) {
      this.setState({
        loadingUser: false,
        error: true
      });
    } else {
      axios
        .get("/finduser", {
          params: {
            username: this.props.match.params.username
          },
          headers: { Authorization: `JWT ${accessString}` }
        })
        .then(response => {
          this.setState({
            loadingUser: false,
            username: response.data.username,
            password: response.data.password,
            error: false
          });
        })
        .catch(error => {
          console.log(error.data);
        });
    }
  }

  handleChange = name => event => {
    this.setState({
      [name]: event.target.value
    });
  };

  updatePassword = e => {
    let accessString = localStorage.getItem("JWT");
    if (accessString === null) {
      this.setState({
        loadingUser: false,
        error: true
      });
    } else {
      e.preventDefault();
      axios
        .put(
          "/updatepassword",
          {
            username: this.state.username,
            password: this.state.password
          },
          {
            headers: { Authorization: `JWT ${accessString}` }
          }
        )
        .then(response => {
          if (response.data.message === "password updated") {
            this.setState({
              updated: true,
              error: false,
              loadingUser: false
            });
          }
        })
        .catch(error => {
          console.log(error.data);
        });
    }
  };

  render() {
    const { username, password, updated, error, loadingUser } = this.state;

    if (error) {
      return (
        <div>
          <HeaderBar title={title} />
          <p style={loading}>
            There was a problem accessing your data. Please go login again.
          </p>
          <LinkButtons
            style={loginButton}
            buttonText={"Go Login"}
            link="/login"
          />
        </div>
      );
    } else if (loadingUser !== false) {
      return (
        <div>
          <HeaderBar title={title} />
          <p style={loading}>Loading user data...</p>
        </div>
      );
    } else if (loadingUser === false && updated === true) {
      return <Redirect to={`/userProfile/${username}`} />;
    } else if (loadingUser === false) {
      return (
        <div>
          <HeaderBar title={title} />
          <form className="profile-form" onSubmit={this.updatePassword}>
            <TextField
              style={inputStyle}
              id="password"
              label="password"
              value={password}
              onChange={this.handleChange("password")}
              type="password"
            />
            <SubmitButtons
              buttonStyle={saveButton}
              buttonText={"Save Changes"}
            />
          </form>
          <LinkButtons
            buttonStyle={homeButton}
            buttonText={"Go Home"}
            link={"/"}
          />
          <LinkButtons
            buttonStyle={cancelButton}
            buttonText={"Cancel Changes"}
            link={`/userProfile/${username}`}
          />
        </div>
      );
    }
  }
}

export default UpdatePassword;
