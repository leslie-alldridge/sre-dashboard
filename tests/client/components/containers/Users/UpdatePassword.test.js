import React from "react";
import { shallow } from "enzyme";
import UpdatePassword from "../../../../../src/containers/Users/UpdatePassword";
describe("MyComponent", () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<UpdatePassword debug />);

    expect(component).toMatchSnapshot();
  });
});
