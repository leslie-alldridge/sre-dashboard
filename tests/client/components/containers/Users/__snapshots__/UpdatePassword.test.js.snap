// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyComponent should render correctly in "debug" mode 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UpdatePassword
    debug={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <HeaderBar
          title={
            Object {
              "pageTitle": "Update Password Screen",
            }
          }
        />,
        <p
          style={
            Object {
              "fontSize": "24px",
              "margin": "1em",
            }
          }
        >
          There was a problem accessing your data. Please go login again.
        </p>,
        <LinkButtons
          buttonText="Go Login"
          link="/login"
          style={
            Object {
              "background": "royalblue",
              "margin": "1em",
              "padding": "1em",
            }
          }
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "title": Object {
            "pageTitle": "Update Password Screen",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "There was a problem accessing your data. Please go login again.",
          "style": Object {
            "fontSize": "24px",
            "margin": "1em",
          },
        },
        "ref": null,
        "rendered": "There was a problem accessing your data. Please go login again.",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "buttonText": "Go Login",
          "link": "/login",
          "style": Object {
            "background": "royalblue",
            "margin": "1em",
            "padding": "1em",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <HeaderBar
            title={
              Object {
                "pageTitle": "Update Password Screen",
              }
            }
          />,
          <p
            style={
              Object {
                "fontSize": "24px",
                "margin": "1em",
              }
            }
          >
            There was a problem accessing your data. Please go login again.
          </p>,
          <LinkButtons
            buttonText="Go Login"
            link="/login"
            style={
              Object {
                "background": "royalblue",
                "margin": "1em",
                "padding": "1em",
              }
            }
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "title": Object {
              "pageTitle": "Update Password Screen",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "There was a problem accessing your data. Please go login again.",
            "style": Object {
              "fontSize": "24px",
              "margin": "1em",
            },
          },
          "ref": null,
          "rendered": "There was a problem accessing your data. Please go login again.",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "buttonText": "Go Login",
            "link": "/login",
            "style": Object {
              "background": "royalblue",
              "margin": "1em",
              "padding": "1em",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
